### DISCUSSION TITLE
How to handle RFC 8482 (ANY request getting a HINFO response)

### DISCUSSION THREAD

**Discussion by EricSoonius** [MichaCo/DnsClient.NET#202] [Q&A] [answered]
For a project I want to resolve host names in code and not using the DNS settings from the OS/IP stack.
Most host names are resolved correctly to an IP address using ANY except ```logc.site24x7.com``` which returns a HINFO response without an A or CNAME record.
https://blog.cloudflare.com/rfc8482-saying-goodbye-to-any/

On the website of DnsClient this DNS query is resolved correctly using type A or CNAME so I would like to know how I should handle RFC 8482 responses with the DnsClient library.

The A and CNAME type returned no results with this code  (using 8.8.8.8:53 in the lookupClient)
```
private static IReadOnlyList<DnsResourceRecord> Query(string host, IDnsQuery lookupClient, QueryType queryType)
{
  var result = lookupClient.Query(host, queryType, queryType);
  return result.Answers;
}
```

This is the concept code I created to query the DNS again
```
private static IReadOnlyList<DnsResourceRecord> Query(string host, IDnsQuery lookupClient, QueryType queryType)
{
  var dnsQuestion = new DnsQuestion(host, queryType, QueryClass.IN);
  var options = new DnsQueryAndServerOptions
  {
    ContinueOnDnsError = true,
    ContinueOnEmptyResponse = true,
    Recursion = true,
    UseCache = false, //for testing disable cache
    UseTcpFallback = true,
    UseTcpOnly = false
  };

  var result = lookupClient.Query(dnsQuestion, options);
  return result.Answers;
}
```
And the last test returned an A but no CNAME record.
So what is the best way to perform a DNS question to handle RFC8482 correctly?

**Comment by MichaConrad** [comment]
Hi Eric,
Did you read [the blog post you posted](https://blog.cloudflare.com/rfc8482-saying-goodbye-to-any/)? 
if actually explains very well why you'll get an HINFO response for many websites hosted by Cloudflare but also others.

TLDR is, ANY is obsolete and should not be used anymore. 

To your question, just don't use it anymore. The blog post also has some examples for different answers which only have the response type in common. 
And, if you want to check if the response is from a domain where the host removed ANY as a valid response, then just check for the existence of a single HINFO record, no matter the content.

**Comment by EricSoonius** [comment, marked as answer]
Yes I read the blog, it explained to me why I was getting no result when using query type ANY.
The funny thing was when I switched in the code to query type A I was getting no result.
A few hairs less, and several cups of coffee later I still could not understand why.
Now a few hours later, I get on both methods mentioned above a result with query type A.
Is cache somewhere playing tricks on me?

Ergo, always use query type A to get the IP address of a host :)

**Reply by MichaConrad** [reply]
> Ergo, always use query type A to get the IP address of a host :)

yup 👍
